api/app/signals/apps/signals/managers.py

```diff
@@ -644,26 +649,77 @@ def update_type(self, data, signal):
 
         return signal_type
 
-    def _update_directing_departments_no_transaction(self, data, signal):
-        from signals.apps.signals.models.directing_departments import DirectingDepartments
+    def _update_user_signal_no_transaction(self, data, signal):
+        from signals.apps.users.models import User, SignalUser
+        try:
+            user_id = data['user_assignment']['user']['id']
+            signal.user_assignment = SignalUser.objects.create(
+                _signal=signal,
+                user=None if not user_id else User.objects.get(pk=user_id),
+                created_by=data['created_by'] if 'created_by' in data else None
+            )
+            signal.save()
+        except Exception:
```

Comment:
When we end up in this `except` clause, no `signal.user_assignment` will not be changed --- but no HTTP 400 returned to the API client either. Is that the intended behavior?

Similar problems with the data are generally raised as exceptions.

Reply:
check